### PR TITLE
fix: decode parameters if needed

### LINKED ISSUE #188: SSH backends choke on bytes as parameters
If bytes are passed as parameters to proxmoxer, they end up formatted like `b'this is a parameter'` instead of `this is a parameter`.  This makes pvesh choke when the actual API call is made.

I have a fix, will create PR immediately after this ticket.

### PR DESCRIPTION
**Pull request by asdorsey**
Fixes #188

### CODE REVIEW

**Comment by jhollowe** [comment]
Please add tests for this new functionality

**Comment by asdorsey** [comment]
Test added, it's basically a copy of `test_request_data` but with the data value in bytes format.

File: tests/test_command_base.py [hunk 1]
```diff
@@ -105,6 +105,20 @@ def test_request_data(self, mock_exec):
             "json",
         ]
 
+    def test_request_bytes_data(self, mock_exec):
+        resp = self._session.request("GET", self.base_url + "/fake/echo", data={"key": b"bytes-value"})
```
Comment (jhollowe) [inline tests/test_command_base.py:109]: Please resolve the formatter/linter CI failures

**Review by jhollowe — CHANGES_REQUESTED** [review]
Looks good except for the needed formatting fix

**Pull request merged**